File: ChoreAssignmentTracker/src/ChoreAssignmentTracker.WebApp/projects/chore-assignment-tracker/src/app/models/assignment.ts
export interface Assignment {
  assignmentId: string;
  choreId: string;
  choreName?: string;
  familyMemberId: string;
  familyMemberName?: string;
  assignedDate: Date;
  dueDate: Date;
  completedDate?: Date;
  isCompleted: boolean;
  isVerified: boolean;
  notes?: string;
  pointsEarned: number;
  isOverdue: boolean;
  createdAt: Date;
}

export interface CreateAssignment {
  choreId: string;
  familyMemberId: string;
  assignedDate: Date;
  dueDate: Date;
  notes?: string;
}

export interface UpdateAssignment {
  dueDate: Date;
  notes?: string;
}

export interface CompleteAssignment {
  notes?: string;
}

export interface VerifyAssignment {
  points: number;
}

File: ChoreAssignmentTracker/src/Assignments.Core/Models/Assignment.cs
namespace Assignments.Core.Models;

public class Assignment
{
    public Guid Id { get; set; }
    public Guid ChoreId { get; set; }
    public Guid FamilyMemberId { get; set; }
    public DateTime DueDate { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime? CompletedAt { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public Guid? TenantId { get; set; }
}
